polkadot/node/core/backing/src/lib.rs

```diff
@@ -224,6 +227,15 @@ struct PerRelayParentState {
 	fallbacks: HashMap<CandidateHash, AttestingData>,
 	/// The minimum backing votes threshold.
 	minimum_backing_votes: u32,
+	/// If true, we're appendindg extra bits in the BackedCandidate validator indices bitfield,
+	/// which represent the assigned core index.
+	inject_core_index: bool,
+	/// The core state for all cores
+	cores: Vec<CoreState>,
+	/// The validator groups at this relay parent.
+	validator_groups: Vec<Vec<ValidatorIndex>>,
+	/// The associated group rotation information.
+	group_rotation_info: GroupRotationInfo,
```

Comment:
Those two are really per session. Storing them per relay parent is rather unnecessary. Seems fine for an interim solution though.